webhook/mutator.go

```diff
@@ -250,3 +267,35 @@ func updateAnnotation(target map[string]string, added map[string]string) []patch
 	}
 	return patch
 }
+
+func makeCopy(src *SideCar) (*SideCar, error) {
+	data, err := yaml.Marshal(src)
+	if err != nil {
+		return nil, err
+	}
+
+	var dst SideCar
+	err = yaml.Unmarshal(data, &dst)
+	if err != nil {
+		return nil, err
+	}
+
+	return &dst, nil
+}
+
+func getEnvToInject(annotations map[string]string) []corev1.EnvVar {
+	var env []corev1.EnvVar
+	sz := len(sideCarNameSpace)
+
+	for key, value := range annotations {
+		if strings.HasPrefix(key, sideCarNameSpace) && len(key) > sz {
```

Comment:
I believe len(key) is more efficient than the prefix lookup. What about inverting the condition?

Reply:
This code runs just once when a deployment is submitted for an app :)

Reply:
@jcchavezs Thanks for the review and comment. It make sense to invert the condition - done.